### DISCUSSION TITLE
Tethys 3 install-ability on M1 macs

### DISCUSSION THREAD

**Discussion by rileyhales** [tethysplatform/tethys#865] [Q&A]
FYI post:

I've discovered that tethys 3 is not installable on the 2021 macbooks which have the M1 Pro chips. Running the usual sequence of install commands will fail to solve the conda environment. The M1 Pro chips look for conda packages under `conda-forge/osx-arm64` and similarly on conda-forge and other channels. Other macs using the standard M1 chip will look under `conda-forge/osx-64` and successfully install. Specifically, django 2.2 and some peer dependencies are not available on the conda-forge channels for the arm64 architecture. Since thats such an old version of django, i don't expect that they would go back and regenerate those packages. I don't see an easy work around, at least for now, but you can successfully install tethys 4 on an M1 Pro mac. That may just be a limitation of the 2021 macbooks.

I'll post more info here if i discover a work around or other relevant information

**Comment by swainn** [comment]
@rileyhales have you tried this recently? Tethys Platform 3.4.4 has been released since this post? Also, could you try this with the [Tethys 4 release candidate](https://github.com/tethysplatform/tethys/discussions/878)?

**Comment by rileyhales** [comment]
(scratch) rchales@macstudio notebook_scratch % mamba install -c conda-forge -c tethysplatform tethys-platform

(mamba logo omitted) ...
      
Looking for: ['tethys-platform']

conda-forge/osx-arm64                                       Using cache
conda-forge/noarch                                          Using cache
pkgs/main/osx-arm64                                           No change
pkgs/main/noarch                                              No change
pkgs/r/osx-arm64                                              No change
pkgs/r/noarch                                                 No change
tethysplatform/osx-arm64                                      No change
tethysplatform/noarch                                         No change

Pinned packages:
  - python 3.9.*


Encountered problems while solving:
  - nothing provides django 2.2.* needed by tethys-platform-3.4.3-pyhd8ed1ab_0

**Reply by rileyhales** [reply]
It didn't find 3.4.4 as the latest on its own but you get the same thing if you try to install `tethys-platform==3.4.4`

```
Encountered problems while solving:
  - nothing provides django 2.2.* needed by tethys-platform-3.4.4-py_0
 ```


It looks like i can manually install django 2.2 from pip now (which didn't seem to work previously, if i remember right). But if you try to conda install tethys platform there is some incompatibility that won't resolve. Conda says that python is pinned at 3.9 and the django docs for 2.2 say that 3.9 is compatible. I haven't tried to figure out where the error comes from

More recent django versions can be installed just fine so tethys 4 is installable.

**Reply by swainn** [reply]
Dang. I was hoping that using a more recent package would mean that it would have solved with dependencies that were built for M1 Macs too. I'm guessing the issue is that since Django 2.2 is no longer supported, there haven't been any recent builds of it. So they don't have a Django 2.2 package for the new architecture. Probably true of some of our other dependencies too I would guess.

Since Tethys 4 has a release candidate out now and will be the official release soon, I think it is reasonable to say that M1 Macs are only supported on Tethys 4+ as you suggested.

Thank you for taking time to test this again.
